cl/search/documents.py

```diff
@@ -539,3 +547,276 @@ def prepare_person_child(self, instance):
 
     def prepare_school(self, instance):
         return [e.school.name for e in instance.educations.all()] or None
+
+
+# RECAP
+class DocketBaseDocument(Document):
+    docket_child = JoinField(relations={"docket": ["recap_document"]})
+    timestamp = fields.DateField()
+
+    class Django:
+        model = Docket
+        ignore_signals = True
+
+    def prepare_timestamp(self, instance):
+        return datetime.utcnow()
+
+
+@recap_index.document
+class RECAPDocument(DocketBaseDocument):
+    id = fields.IntegerField(attr="pk")
+    docket_entry_id = fields.IntegerField(attr="docket_entry.pk")
+    description = fields.TextField(
+        attr="docket_entry.description",
+        analyzer="text_en_splitting_cl",
+        fields={
+            "exact": fields.TextField(
+                attr="docket_entry.description", analyzer="english_exact"
+            ),
+        },
+        search_analyzer="search_analyzer",
+    )
+    entry_number = fields.IntegerField(attr="docket_entry.entry_number")
+    entry_date_filed = fields.DateField(attr="docket_entry.date_filed")
+    entry_date_filed_text = fields.TextField(
+        analyzer="text_en_splitting_cl",
+        fields={
+            "exact": fields.TextField(analyzer="english_exact"),
+        },
+        search_analyzer="search_analyzer",
+    )
+    short_description = fields.TextField(
+        attr="description",
+        analyzer="text_en_splitting_cl",
+        fields={
+            "exact": fields.TextField(
+                attr="description", analyzer="english_exact"
+            ),
+        },
+        search_analyzer="search_analyzer",
+    )
+    document_type = fields.TextField(
+        analyzer="text_en_splitting_cl",
+        fields={
+            "exact": fields.TextField(analyzer="english_exact"),
+        },
+        search_analyzer="search_analyzer",
+    )
+    document_number = fields.IntegerField(attr="document_number")
+    pacer_doc_id = fields.KeywordField(attr="pacer_doc_id")
+    plain_text = fields.TextField(
+        attr="plain_text",
+        analyzer="text_en_splitting_cl",
+        fields={
+            "exact": fields.TextField(
+                attr="plain_text", analyzer="english_exact"
+            ),
+        },
+        search_analyzer="search_analyzer",
+    )
+    attachment_number = fields.IntegerField(attr="attachment_number")
+    is_available = fields.BooleanField(attr="is_available")
+    page_count = fields.IntegerField(attr="page_count")
+    filepath_local = fields.KeywordField(index=False)
+
+    class Django:
+        model = RECAPDocument
+        ignore_signals = True
+
+    def prepare_document_type(self, instance):
+        return instance.get_document_type_display()
+
+    def prepare_local_path(self, instance):
+        if instance.filepath_local:
+            if not instance.filepath_local.storage.exists(
+                instance.filepath_local.name
```

Comment:
Do we want to hit S3 for every file we index? That's 300M+ S3 calls to do the whole thing?

Reply:
Yeah, I think the alternative is to index the file path even if it doesn't exists in S3 anymore, is that ok?

Reply:
I'm actually not sure we need the field? Is it in Solr?

Reply:
I confirmed that this field is also available in Solr, specifically to be shown in the Search API. So I kept it and now we're only indexing the name stored in DB `instance.filepath_local.name` without checking if it exists in S3 or not.